packages/block-library/src/list-item/edit.js

```diff
@@ -1,47 +1,210 @@
+/**
+ * External dependencies
+ */
+import { get, first } from 'lodash';
+
 /**
  * WordPress dependencies
  */
 import {
 	RichText,
 	useBlockProps,
 	useInnerBlocksProps,
+	BlockControls,
+	store as blockEditorStore,
 } from '@wordpress/block-editor';
-import { __ } from '@wordpress/i18n';
+import { isRTL, __, _x } from '@wordpress/i18n';
 import { createBlock } from '@wordpress/blocks';
+import { useDispatch, useSelect, select } from '@wordpress/data';
+import { ToolbarButton } from '@wordpress/components';
+import {
+	formatOutdent,
+	formatOutdentRTL,
+	formatIndentRTL,
+	formatIndent,
+} from '@wordpress/icons';
+
+function IdentUI( { attributes, clientId } ) {
+	const { canOutdent, blockIndex } = useSelect(
+		( innerSelect ) => {
+			const { getBlockRootClientId, getBlockIndex } = innerSelect(
+				blockEditorStore
+			);
+			const grandParentId = getBlockRootClientId(
+				getBlockRootClientId( clientId )
+			);
+			return {
+				blockIndex: getBlockIndex( clientId ),
+				canOutdent: !! grandParentId,
+			};
+		},
+		[ clientId ]
+	);
+	const { replaceBlocks } = useDispatch( blockEditorStore );
+	return (
+		<>
+			<ToolbarButton
+				icon={ isRTL() ? formatOutdentRTL : formatOutdent }
+				title={ __( 'Outdent' ) }
+				describedBy={ __( 'Outdent list item' ) }
+				shortcut={ _x( 'Backspace', 'keyboard key' ) }
+				disabled={ ! canOutdent }
+				onClick={ () => {
```

Comment:
Should we try to decouple the logic from the UI. Maybe a `const { indentListItem } = useListActions( clientId )` or `const indentListItem = useListIndent( clientId }` or something else. The idea is to make easier to reuse for keyboard shortcuts as well?